default/AI/AIstate.py

```diff
@@ -268,6 +273,39 @@ def assess_planet_threat(self, pid, sightingAge=0):
         infra = 0  # doesn't really contribute to combat since damage to shields & 'defense' done first
         return {'overall': defense*(defense + shields + infra), 'attack': defense, 'health': (defense + shields + infra)}
 
+    def log_immobile_fleet(self, fleet, system, rating):
+        if not fleet or not system:
+            return
+        universe = fo.getUniverse()
+        print "Immobile fleet %s at %s has rating %.1f and %d ships:" % (fleet, system, rating.get('overall', 0), len(fleet.shipIDs))
+        for ship_id in fleet.shipIDs:
```

Comment:
You can collect all designs to `collections.Counter` with key `this_ship.designID` to stack same designs.